diff --git a/dist/components/ChartAnalytics/DHIS2Chart.js b/dist/components/ChartAnalytics/DHIS2Chart.js
index d430ca74ae86b62f32a88f3530cca61275264e81..f150bce54a376d8616917ac4c02d0a43b654a8c3 100644
--- a/dist/components/ChartAnalytics/DHIS2Chart.js
+++ b/dist/components/ChartAnalytics/DHIS2Chart.js
@@ -45,7 +45,7 @@ function ChartAnalyticsComponent({
   config,
   containerProps,
   setRef
-}) {
+}, ref) {
   const id = react.useRef(`${dhis2Utils.uid()}-chart-item`);
   const { chart } = useChart_js.useChart({ id: id.current, analytics, config });
   if (!chart) {
@@ -55,7 +55,7 @@ function ChartAnalyticsComponent({
     HighchartsReact__default.default,
     {
       immutable: true,
-      ref: setRef,
+      ref: ref ?? setRef,
       containerProps: {
         id: id.current,
         style: { width: "100%", height: "100%" },
diff --git a/dist/esm/components/ChartAnalytics/DHIS2Chart.js b/dist/esm/components/ChartAnalytics/DHIS2Chart.js
index dbc243d78769427c207bac62c26fc1723287ac1f..0bab49b39fff6465e0a0ff87904a9608bf85708f 100644
--- a/dist/esm/components/ChartAnalytics/DHIS2Chart.js
+++ b/dist/esm/components/ChartAnalytics/DHIS2Chart.js
@@ -20,7 +20,7 @@ function ChartAnalyticsComponent({
   config,
   containerProps,
   setRef
-}) {
+}, ref) {
   const id = useRef(`${uid()}-chart-item`);
   const { chart } = useChart({ id: id.current, analytics, config });
   if (!chart) {
@@ -30,7 +30,7 @@ function ChartAnalyticsComponent({
     HighchartsReact,
     {
       immutable: true,
-      ref: setRef,
+      ref: ref ?? setRef,
       containerProps: {
         id: id.current,
         style: { width: "100%", height: "100%" },
